addons/web/static/src/legacy/frontend/rainbowman/rainbow_man.js

```diff
@@ -61,7 +63,15 @@ var RainbowMan = Widget.extend({
                 }, 600); // destroy only after fadeout animation is completed
             }, this.delay);
         }
-        this.$('.o_reward_msg_content').append(this.options.message);
+        let message = this.options.message;
+        if (!this.options.messageIsHtml) {
+            if (message instanceof jQuery) {
+                message = message.text();
+            } else if (message instanceof Element) {
+                message = message.textContent;
+            }
+        }
+        this.$('.o_reward_msg_content').append(message);
```

Comment:
That's definitely not correct, the parameter to `append` is interpreted as DOM content, so if it's a string it's going to be treated as [an `htmlString`](https://api.jquery.com/Types/#htmlString).

To ensure jQuery treats it as text (while still being able to *append* it), it needs to [be wrapped in a `Text` node](https://developer.mozilla.org/en-US/docs/Web/API/Document/createTextNode).